netket/stats/mc_stats.py

```diff
@@ -98,42 +101,87 @@ def __getattr__(self, name):
         elif name in ("variance", "Variance"):
             return self.variance
         elif name in ("error_of_mean", "Sigma"):
-            return self.error_of_mean
+            return self.error
         elif name in ("R_hat", "R"):
-            return self.R_hat
+            return self.r_hat
         elif name in ("tau_corr", "TauCorr"):
             return self.tau_corr
         else:
             raise AttributeError(
                 "'Stats' object object has no attribute '{}'".format(name)
             )
 
+    @property
+    def n_chains(self):
+        return self._μ.size * mpi.n_nodes
 
-def _get_blocks(data, block_size):
-    chain_length = data.shape[1]
+    @struct.property_cached(pytree_node=True)
+    def mean(self) -> float:
+        return _mean(self._μ)
 
-    n_blocks = int(np.floor(chain_length / float(block_size)))
+    @struct.property_cached(pytree_node=True)
+    def variance(self) -> float:
+        return _mean(self._μ2) - self.mean ** 2
 
-    return data[:, 0 : n_blocks * block_size].reshape((-1, block_size)).mean(axis=1)
+    @struct.property_cached(pytree_node=True)
+    def error(self) -> float:
+        return jnp.sqrt(self.batch_variance / self.n_chains)
 
+    @struct.property_cached(pytree_node=True)
+    def r_hat(self) -> float:
+        if self.n_chains > 1:
+            return jnp.sqrt(
+                (self._n - 1) / self._n + self.batch_variance / self.variance
+            )
+        else:
+            return jnp.nan
+
+    @struct.property_cached(pytree_node=True)
+    def tau_corr(self) -> float:
+        return jnp.clip(0.5 * (self._n * self.batch_variance / self.variance - 1), 0)
+
+    @struct.property_cached(pytree_node=True)
+    def batch_variance(self) -> float:
+        return _var(self._μ)
+
+    @struct.property_cached(pytree_node=True)
+    def block_variance(self) -> float:
+        return _mean(self._μ2_w) - self.mean ** 2
+
+    @property
+    def block_size(self):
+        return self._w
+
+    def merge(self, other):
```

Comment:
I would call this just the operator +, not merge, also in anticipation of support for other operations

Reply:
(incidentally, rescaling and subtraction are easy to do at this point)

Reply:
ah wait, this is actually a merge... sorry

Reply:
I don't think it's really a sum.
While if we were tracking only variance and mean that would be it, for what regards autocorrelation and other things.. I'm not so sure.

Reply:
yes I agree, It is a merge operation, I realized it afterwards